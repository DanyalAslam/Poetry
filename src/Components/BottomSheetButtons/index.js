import React from 'react'
import { Text, View, Image } from 'react-native'
import RippleTouch from '../../Components/RippleTouch/index.js'
import fonts from '../../assets/fonts/index.js'
import { vh, vw } from '../../Units/index.js'
import { appTheme } from '../../Utils/index.js'


class BottomSheetButtons extends React.Component {

    render() {

        return (
            <RippleTouch style={{

                width: 100 * vw,
                paddingLeft: 5 * vw,
                paddingRight: 5 * vw,


            }}
            onPress={this.props.onPress}
            >
                <View style={{
                    flexDirection: 'row',
                    alignItems: 'center',
                    borderBottomWidth: 0.1 * vw,
                    paddingVertical: 1.5 * vh,
                    borderBottomColor: appTheme.grayish
                }}>


                    <Image
                        source={this.props.source}
                        style={{ width: 5 * vw, height: 5 * vw, resizeMode: 'contain' }}
                    />
                    <Text style={{
                        fontFamily: fonts.SSB,
                        fontSize: 2 * vh,
                        marginLeft: 3 * vw
                    }}>
                      {this.props.text}
                        </Text>

                </View>

            </RippleTouch>

        );
    }
}

export default BottomSheetButtons;