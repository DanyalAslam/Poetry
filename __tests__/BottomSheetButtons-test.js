/**
 * @format
 */

import 'react-native';
import React from 'react';
import BottomSheetButtons from '../src/Components/BottomSheetButtons';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const tree = renderer.create(<BottomSheetButtons />).toJSON();
    expect(tree).toMatchSnapshot();
  });
 