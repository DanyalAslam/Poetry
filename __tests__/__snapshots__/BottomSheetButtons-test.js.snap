// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={
    Object {
      "paddingLeft": 60,
      "paddingRight": 60,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 0.75,
        "flexDirection": "row",
        "paddingVertical": 20.009999999999998,
      }
    }
  >
    <Image
      style={
        Object {
          "height": 37.5,
          "resizeMode": "contain",
          "width": 37.5,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "SourceSerifPro-SemiBold",
          "fontSize": 26.68,
          "marginLeft": 22.5,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  />
</View>
`;
